src/components/application_manager/test/commands/mobile/show_constant_tbt_test.cc

```diff
@@ -0,0 +1,402 @@
+
+    msg_params[am::hmi_request::navi_texts][index]
+              [am::hmi_request::field_name] = static_cast<int32_t>(
+                  hmi_apis::Common_TextFieldName::navigationText2);
+    msg_params[am::hmi_request::navi_texts][index++]
+              [am::hmi_request::field_text] = kNavigationText2;
+
+    msg_params[am::hmi_request::navi_texts][index]
+              [am::hmi_request::field_name] =
+                  static_cast<int32_t>(hmi_apis::Common_TextFieldName::ETA);
+    msg_params[am::hmi_request::navi_texts][index++]
+              [am::hmi_request::field_text] = kEta;
+
+    msg_params
+        [am::hmi_request::navi_texts][index][am::hmi_request::field_name] =
+            static_cast<int32_t>(hmi_apis::Common_TextFieldName::totalDistance);
+    msg_params[am::hmi_request::navi_texts][index++]
+              [am::hmi_request::field_text] = kTotalDistance;
+
+    msg_params[am::hmi_request::navi_texts][index]
+              [am::hmi_request::field_name] = static_cast<int32_t>(
+                  hmi_apis::Common_TextFieldName::timeToDestination);
+    msg_params[am::hmi_request::navi_texts][index++]
+              [am::hmi_request::field_text] = kTimeToDestination;
+
+    (*msg)[am::strings::msg_params] = msg_params;
+
+    return msg;
+  }
+
+  void ExpectedManageMobileCommandResultCode(mobile_apis::Result::eType code) {
+    EXPECT_CALL(app_mngr_, ManageMobileCommand(MobileResultCodeIs(code), _))
+        .WillOnce(Return(true));
+  }
+
+ private:
+  void SetUp() OVERRIDE {
+    message_ = CreateMessageWithParams();
+    expected_message_ = CreateExpectedMessage();
+    mock_app_ = CreateMockApp();
+    ON_CALL(app_mngr_, application(kConnectionKey))
+        .WillByDefault(Return(mock_app_));
+    ON_CALL(app_mngr_, GetPolicyHandler())
+        .WillByDefault(ReturnRef(mock_policy_handler_interface_));
+  }
+
+ protected:
+  MessageSharedPtr message_;
+  SharedPtr<ShowConstantTBTRequest> request_;
+  MessageSharedPtr expected_message_;
+  MockAppPtr mock_app_;
+  MockPolicyHandlerInterface mock_policy_handler_interface_;
+};
+
+TEST_F(ShowConstantTBTRequestTest, Run_AllCorrectData_SendHMIRequest) {
+  am::ApplicationConstSharedPtr app_const_ptr(mock_app_);
+  EXPECT_CALL(mock_message_helper_,
+              ProcessSoftButtons(
+                  (*message_)[am::strings::msg_params], app_const_ptr, _, _))
+      .WillOnce(Return(mobile_apis::Result::SUCCESS));
+
+  EXPECT_CALL(mock_message_helper_, VerifyImage(_, app_const_ptr, _))
+      .Times(2)
+      .WillRepeatedly(Return(mobile_apis::Result::SUCCESS));
+
+  EXPECT_CALL(*mock_app_, app_id()).WillOnce(Return(kAppId));
+
+  am::ApplicationSharedPtr app_ptr(mock_app_);
+  EXPECT_CALL(mock_message_helper_,
+              SubscribeApplicationToSoftButton(_, app_ptr, _));
+
+  smart_objects::SmartObject msg_params;
+  EXPECT_CALL(*mock_app_, set_tbt_show_command(_))
+      .WillOnce(SaveArg<0>(&msg_params));
+
+  MessageSharedPtr result;
```

Comment:
@BSolonenko please give this variable more appropriate name. This is generated by SDL hmi command.